### ISSUE TITLE
Add a test for `Document.add_root(not_a_component)`

### ISSUE THREAD

**Issue by havocp** [bokeh/bokeh#3117]
For use with `get_model_by_name` for example we were talking about having data sources as roots. (See https://github.com/bokeh/bokeh/pull/3112 ). The old 0.10 document only allowed Component subtypes at the root; we may still have checks for that, or `embed.coffee` may try to render all roots even if they aren't "renderable". So we should test (and fix if necessary) having a data source or other non-renderable model as a root.

**Comment by havocp** [comment]
This may fail here, because there's no `default_view`: https://github.com/bokeh/bokeh/blob/57203a7668687aacf92c47795affd7da0c83d017/bokehjs/src/coffee/common/embed.coffee#L11-L14

If so all we have to do is check here whether there's a `default_view` and skip rendering any root that lacks one:
https://github.com/bokeh/bokeh/blob/57203a7668687aacf92c47795affd7da0c83d017/bokehjs/src/coffee/common/embed.coffee#L40-L50
be sure to catch both uses of `render_model`, or put the check inside `render_model`.

**Comment by canavandl** [comment]
@havocp is correct that adding non-components to the document root will result in an error related to the model not having a ``default_view`` method.

```
Error rendering Bokeh items  TypeError: model.default_view is not a function
```

**Comment by havocp** [comment]
I think the right behavior is to skip the model instead of throwing that error if there's no default_view

**Comment by canavandl** [comment]
Agreed - PR coming

**Comment by p-himik** [comment]
The referenced PR is closed as superseded by #3703, but in the end, the PR doesn't solve the issue. Right now, there's no way to add a data source to a document without attaching it to some model.

@canavandl Can you recall, why the commit https://github.com/bokeh/bokeh/pull/3703/commits/3d64e9e41adbe8dd7f37a503a2029a4f6f9a9ca9 was made?

**Comment by canavandl** [comment]
@p-himik I'm sorry, I don't recall why I made that change.

**Comment by mattpap** [comment]
@p-himik, this will be resolved in PR #6340.

**Comment by bryevdv** [comment]
To implement #3674 it will explicitly be necessary to make it possible to add non-view models as roots:

    curdoc().add_root(some_data_model)

I propose to simply allow any object to added as a root in that work, so I will close this issue and simply leave a note there.
